packages/cli-api/src/main/ts/executors/deprecate.ts

```diff
@@ -0,0 +1,72 @@
+import { IDeprecatePackageParams, NpmRegClientWrapper, INpmRegClientWrapper, RegClient } from '@qiwi/npm-batch-client'
+
+import { defaultRateLimit } from '../default'
+import { TDeprecationConfig } from '../interfaces'
+import { printResults, printResultsJson, withRateLimit } from '../utils'
+
+export const performDeprecation = async (
+  config: TDeprecationConfig,
+  customBatchClient?: INpmRegClientWrapper
+): Promise<void> => {
+  const regClient = new RegClient()
+  const batchClient = customBatchClient || new NpmRegClientWrapper(
+    config.registryUrl,
+    config.auth,
+    withRateLimit<RegClient>(regClient, config.batch?.ratelimit || defaultRateLimit, ['deprecate'])
+  )
+
+  return batchClient
+      .deprecateBatch(config.data)
+      .then(data => processResults(handleSettledResults(data), config))
+}
+
+export const processResults = (results: any[], config: TDeprecationConfig): void => {
+  const enrichedResults = enrichResults(results, config.data)
+  const successfulResults = getSuccessfulResults(enrichedResults)
+  const failedResults = getFailedResults(enrichedResults)
+
+  if (config.batch?.jsonOutput) {
+    printResultsJson(
+      successfulResults,
+      failedResults
+    )
+    return
+  }
+
+  printResults(
+    successfulResults,
+    failedResults,
+    ['packageName', 'version', 'message'],
+    ['packageName', 'version', 'message', 'error'],
+    'Following packages are deprecated successfully:',
+    'Following packages are not deprecated due to errors:'
+  )
+}
+
+type TEnrichedBatchResult = {
+  result: any
+  packageInfo: IDeprecatePackageParams
+}
+
+export const enrichResults = (normalizedResults: any[], data: IDeprecatePackageParams[]): TEnrichedBatchResult[] =>
```

Comment:
I suggest to introduce smth like `INormalizedResult`

Reply:
oh, it is old parameter name, I will rename it to `results`